core/domain/exp_services.py

```diff
@@ -830,20 +831,37 @@ def update_exploration(
             'Commit messages for non-suggestions may not start with \'%s\'' %
             feconf.COMMIT_MESSAGE_ACCEPTED_SUGGESTION_PREFIX)
 
-    exploration = apply_change_list(exploration_id, change_list)
-    _save_exploration(committer_id, exploration, commit_message, change_list)
+    old_exploration = exp_fetchers.get_exploration_by_id(exploration_id)
+    updated_exploration = apply_change_list(exploration_id, change_list)
+    _save_exploration(
+        committer_id, updated_exploration, commit_message, change_list)
 
     discard_draft(exploration_id, committer_id)
     # Update summary of changed exploration.
-    update_exploration_summary(exploration.id, committer_id)
-    user_services.add_edited_exploration_id(committer_id, exploration.id)
+    update_exploration_summary(updated_exploration.id, committer_id)
+    user_services.add_edited_exploration_id(
+        committer_id, updated_exploration.id)
     user_services.record_user_edited_an_exploration(committer_id)
 
-    if (not rights_manager.is_exploration_private(exploration.id) and
+    if (not rights_manager.is_exploration_private(updated_exploration.id) and
             committer_id != feconf.MIGRATION_BOT_USER_ID):
         user_services.update_first_contribution_msec_if_not_set(
             committer_id, utils.get_current_time_in_millisecs())
 
+    if opportunity_services.is_curated_exploration(exploration_id):
+        old_content_count = old_exploration.get_content_count()
+        old_translation_count = old_exploration.get_translation_count()
+        new_content_count = updated_exploration.get_content_count()
+        new_translation_count = updated_exploration.get_translation_count()
+        complete_translation_language_list = (
+            updated_exploration.get_languages_with_complete_translation())
+
+        if old_content_count != new_content_count or old_translation_count != (
+                new_translation_count):
+            opportunity_services.update_exploration_opportunity_with_new_exploration( # pylint: disable=line-too-long
```

Comment:
update_exploration_contribution_opportunities

And also, I'm a bit unsure about this. Is it possible for old/new content count to be the same and old/new translation count to be the same, but the contribution opportunities to have changed? E.g. if you add one and delete one, or something like that? In general I think it might be more robust to replace all of this with a call to opportunity_services.regenerate_exploration_contribution_opportunities() which just updates the whole model from scratch.

Reply:
> Is it possible for old/new content count to be the same and old/new translation count to be the same, but the contribution opportunities to have changed? E.g. if you add one and delete one, or something like that? 

No, I don't think we have to change the OpportunityModel in this case.

>  In general I think it might be more robust to replace all of this with a call to opportunity_services.regenerate_exploration_contribution_opportunities() which just updates the whole model from scratch.

ExplorationOpportunityModels are bind with StoryModel, TopicModel and ExplorationModel out of these 3 I think Exploration is going to be changed frequently. Having these check will help avoid making multiple db calls in most cases. (After publishing exploration mostly we fix typos, do exploration metadata changes etc.)

(I'm okay with making it `regenerate_exploration_contribution_opportunities`, should I do it?)